sweep/sweeper.go

```diff
@@ -1631,6 +1665,81 @@ func (s *UtxoSweeper) ListSweeps() ([]chainhash.Hash, error) {
 	return s.cfg.Store.ListSweeps()
 }
 
+func (s *UtxoSweeper) handleDeadlineAwareInput(
+	input *sweepInputMessage, currentHeight int32) error {
+
+	op := input.input.OutPoint()
+
+	// Get the initial conf target.
+	initialConfTarget := input.params.Fee.ConfTarget
+	if initialConfTarget == 0 {
+		return ErrConfTargetNotSet
+	}
+
+	// Get the initial fee rate.
+	initialFeeRate, err := s.feeRateForPreference(input.params.Fee)
+	if err != nil {
+		return err
+	}
+
+	// Store the fee bump params.
+	params := &feeBumpParam{
+		strategy:          StrategyDeadlineAware,
+		initialConfTarget: initialConfTarget,
+		deadline:          uint32(currentHeight) + initialConfTarget,
+		initialFeeRate:    initialFeeRate,
+	}
+	s.deadlineAwareInputs.Store(op, params)
+
+	log.Tracef("Stored deadline aware input %v", op)
+
+	return nil
+}
+
+func (s *UtxoSweeper) handleFeeBump(currentHeight uint32) {
```

Comment:
IMO this should only be for the deadline-aware inputs. The logic is a bit weird, but I think we can improve that later. We need to make sure that deadline-aware inputs don't stop after MaxSweepAttempts since they are going to be fee-bumping more since they are being re-submitted more in this function